drivers/at25xxx/at25xxx.c

```diff
@@ -120,18 +133,32 @@ int at25xxx_write(const at25xxx_t *dev, uint32_t pos, const void *data, size_t l
         return -ERANGE;
     }
 
+    /* page size is always a power of two */
+    const uint32_t page_shift = 32 - __builtin_clz(PAGE_SIZE) - 1;
+    const uint32_t page_mask = PAGE_SIZE - 1;
+
+    uint32_t page   = pos >> page_shift;
+    uint32_t offset = pos & page_mask;
+
     getbus(dev);
```

Comment:
Now the calls to `getbus()` and `spi_release()` are obsolete here, if `at25xxx_write_page()` also calls them.

Reply:
Good catch, I was meaning to use the internal `_at25xxx_write_page()` function instead.